modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/mechas.dm

```diff
@@ -506,523 +571,260 @@
+	addtimer(CALLBACK(src, PROC_REF(beglaser_4), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_4(atom/target, var/next_cycle)
+	downone_rightfour(target)
+	downtwo_rightfour(target)
 	downthree_rightfour(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_5), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_5(atom/target, var/next_cycle)
+	downthree_rightfour(target)
+	downfour_rightfour(target)
 	downfour_rightthree(target)
-	sleep(10)
-	downfour(target)
-	downfour_leftone(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_6), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_6(atom/target, var/next_cycle)
+	downfour_rightthree(target)
+	downfour_righttwo(target)
 	downfour_rightone(target)
-	sleep(10)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_7), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_7(atom/target, var/next_cycle)
+	downfour_rightone(target)
+	downfour(target)
+	downfour_leftone(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_8), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_8(atom/target, var/next_cycle)
+	downfour_leftone(target)
+	downfour_lefttwo(target)
+	downfour_leftthree(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_9), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_9(atom/target, var/next_cycle)
+	downfour_leftfour(target)
+	downthree_leftfour(target)
+	downtwo_leftfour(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_10), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_10(atom/target, var/next_cycle)
+	downtwo_leftfour(target)
+	downone_leftfour(target)
+	leftfour(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_11), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_11(atom/target, var/next_cycle)
 	leftfour(target)
 	upone_leftfour(target)
-	downone_leftfour(target)
-	sleep(10)
+	uptwo_leftfour(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_12), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_12(atom/target, var/next_cycle)
+	uptwo_leftfour(target)
+	upthree_leftfour(target)
+	upfour_leftfour(target)
+	addtimer(CALLBACK(src, PROC_REF(beglaser_13), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+
+/mob/living/simple_mob/mechanical/mecha/eclipse/proc/beglaser_13(atom/target, var/next_cycle)
 	upfour_leftfour(target)
 	upfour_leftthree(target)
-	upfour_leftthree(target)
-	attackcycle = 3
+	upfour_lefttwo(target)
+	if(prob(50))
+		addtimer(CALLBACK(src, PROC_REF(beglaser), target, next_cycle), 1 SECOND, TIMER_DELETE_ME)
+	else
```

Comment:
empty else?

Reply:
ac k, I am dumb. Thank you for catching that.